### PR TITLE
optimize: the problem of type inconsistency caused by deserialization of BusinessActionContext

### LINKED ISSUE #5011: TCC模式中，BusinessActionContext中反序列化出来的数据类型有问题，比如long类型参数反序列化出来是Integer类型
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
TCC模式中，资源预留接口中，有一个long类型参数，但是在提交或回滚时从BusinessActionContext中取出来的参数类型是Integer类型，这个问题是因为BusinessActionContext反序列化时没有保证与序列化时的数据类型一致。

这个问题我去看看能不能从fastjson上找到解决方案。

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- Seata version:  
- OS :
- Others:

### LINKED ISSUE #5334: TCC commit和rollback方法去除FastJson依赖
## Why you need it?

TCC commit和rollback方法如果有BusinessActionContext类型的入参，要想使用BusinessActionContext中的actionContext变量，则必须使用FastJson的JsonObject。我们项目中并不想依赖FastJson。

## case

image

### PR DESCRIPTION
**Pull request by zw201913**
…zation of BusinessActionContext

- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?

fixes #5011
fixes #5334

### CODE REVIEW

**Review by a364176773 — COMMENTED** [review]
Please add test cases

**Comment by zw201913** [comment]
get it

File: tcc/src/main/java/io/seata/rm/tcc/serializer/ContextSerializerFactory.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * tcc BusinessActionContext serializer factory
+ *
+ * @author zouwei
+ */
+public class ContextSerializerFactory {
+
+    private ContextSerializerFactory() {}
+
+    // ContextSerializer cache
+    private static final ConcurrentMap<String, ContextSerializer> INSTANCES = new ConcurrentHashMap<>();
```
Comment (a364176773) [inline tcc/src/main/java/io/seata/rm/tcc/serializer/ContextSerializerFactory.java:31]: private static final Map<String, ContextSerializer> INSTANCES = new ConcurrentHashMap<>();

**Comment by a364176773** [comment]
#5352

**Pull request closed without merging**